pkg/workload/resources.go

```diff
@@ -124,35 +128,34 @@ func AdjustResources(ctx context.Context, cl client.Client, wl *kueue.Workload)
 
 // ValidateResources validates that requested resources are less or equal
 // to limits.
-func ValidateResources(wi *Info) error {
-	podsetsPath := field.NewPath("podSets")
+func ValidateResources(wi *Info) field.ErrorList {
+	podSetsPath := field.NewPath("podSets")
 	// requests should be less than limits.
-	var allReasons []string
+	var allErrors field.ErrorList
 	for i := range wi.Obj.Spec.PodSets {
 		ps := &wi.Obj.Spec.PodSets[i]
-		psPath := podsetsPath.Child(ps.Name)
+		psPath := podSetsPath.Index(i).Child(ps.Name)
```

Comment:
I'm on the fence about whether or not we should fix this in a separate PR and then cherry-pick it to release branches.
WDYT? @mimowo

Reply:
I extracted this to https://github.com/kubernetes-sigs/kueue/pull/4218

Reply:
TBH, I would prefer to combine them again, so that we can see the impact of the fix on the paths. It will also be easier to cherry-pick them (as one PR vs two). Otherwise we need to cherry-pick without tests.

Reply:
As I leaving comment https://github.com/kubernetes-sigs/kueue/pull/4218#issuecomment-2650458851, I will combine those into this PR, then close #4218